### PR TITLE
[BPK-2410] Theming for featured and destructive buttons

### PR DESCRIPTION
**Pull request by ojcurtis**
+ [x] Check this if you have read and followed the [contributing guidelines](https://github.com/Skyscanner/backpack/blob/master/CONTRIBUTING.md)

_If you are curious about how we review, please read through the [code review guidelines](https://github.com/Skyscanner/backpack/blob/master/CODE_REVIEW_GUIDELINES.md)_

### CODE REVIEW

**Comment by ojcurtis** [comment]
Images for reference. Please note the third button has **hover** active

**Featured Buttons:**
Screenshot 2019-05-02 at 17 09 51

**Destructive Buttons:**
Screenshot 2019-05-02 at 17 05 23

**Comment by ojcurtis** [comment]
One point to note that requires investigation: Storybook theme gets applied, however the docs site is not picking up the theme

File: packages/bpk-mixins/src/mixins/_buttons.scss [hunk 1]
```diff
@@ -152,17 +152,28 @@
...
-  color: $bpk-button-destructive-color;
-  box-shadow: $bpk-button-destructive-box-shadow;
+  @include bpk-themeable-property(color, --bpk-button-destructive-text-color, $bpk-button-destructive-color);
+  @include bpk-themeable-property(background-color, --bpk-button-destructive-background-color, $bpk-button-destructive-background-color);
+  @include bpk-border-lg($bpk-button-destructive-border-color);
+  @include bpk-border-lg(var(--bpk-button-destructive-border-color, $bpk-button-destructive-border-color));
 
   @include bpk-hover {
-    color: $bpk-button-destructive-hover-color;
+    background-image: $bpk-button-destructive-hover-background-image;
```
Comment (k0nserv) [inline packages/bpk-mixins/src/mixins/_buttons.scss]: Why was this introduced?

Comment (ojcurtis) [reply packages/bpk-mixins/src/mixins/_buttons.scss]: It was taken from the secondary bpk-button, so assumed this was needed to be the same

File: UNRELEASED.md [hunk 2]
```diff
@@ -2,6 +2,20 @@
 
 > Place your changes below this line.
 
+**Breaking:**
+- `bpk-tokens`:
+  - Deprecated `BUTTON_FEATURED_BACKGROUND_IMAGE` to allow theming as no longer required.
```
[the remark this answers is not in the document]
Comment (ojcurtis) [reply UNRELEASED.md]: 👍

File: UNRELEASED.md [hunk 3]
```diff
@@ -2,6 +2,20 @@
...
+
+**Added:**
+- `bpk-tokens`:
+  - Added new tokens for destructive and featured buttons to allow theming.
+- `bpk-mixins`:
```
[the remark this answers is not in the document]
Comment (ojcurtis) [reply UNRELEASED.md]: 👍

File: packages/bpk-mixins/src/mixins/_buttons.scss (hunk 1 above)
Comment (ojcurtis) [reply packages/bpk-mixins/src/mixins/_buttons.scss]: Looks like it makes no difference so we can remove it

**Pull request merged**